perun-utils/password-manager/perun.passwordManager

```diff
@@ -45,15 +54,25 @@ function change() {
 	case $TYPE in
 	KERBEROS_HEIMDAL)
 		# Change password
-		RET=`/usr/bin/kadmin.heimdal --principal=$ADMINPRINCIPAL --keytab=$KEYTAB --realm=$REALM passwd --password="${PASSWORD}" ${USERLOGIN}@${REALM} 2>&1`
-		if [ $? -ne 0 ]; then
+		RET=$(timeout -k $TIMEOUT_KILL $TIMEOUT /usr/bin/kadmin.heimdal --principal=$ADMINPRINCIPAL --keytab=$KEYTAB --realm=$REALM passwd --password="${PASSWORD}" ${USERLOGIN}@${REALM} 2>&1)
+		RET_CODE=$?
+		if [ $RET_CODE -eq 124 ]; then
+			logger "Perun-PasswordChange: setting new password timed out for $LOGINNAMESPACE:$USERLOGIN in $REALM (Warning: this error can mask original error 124 from KDC!)"
```

Comment:
It would be nice to log content of $RET too. Just in case, kadmin command ends with return code 124 (without timeout).

Reply:
Fixed with next commit.